AK/IntrusiveList.h

```diff
@@ -273,7 +273,11 @@ inline typename IntrusiveList<T, member>::Iterator IntrusiveList<T, member>::end
 template<class T, IntrusiveListNode T::*member>
 inline T* IntrusiveList<T, member>::node_to_value(IntrusiveListNode& node)
 {
-    return (T*)((char*)&node - ((char*)&(((T*)nullptr)->*member) - (char*)nullptr));
+// FIXME: WTF is even going on here, why are we casting nullptrs?
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wold-style-cast"
+    return reinterpret_cast<T*>((char*)&node - ((char*)&(((T*)nullptr)->*member) - (char*)nullptr));
```

Comment:
This is a hand rolled `offsetof`. I believe we can use `__builtin_offsetof` instead, to be more standards compliant.

Reply:
Actually it's not that simple. I don't know how to implement this particular operation with the pointer-to-member using offsetof. But the comment should be more like `// arcane magic here, get pointer to T given the list node pointer-to-member in the CRTP'd type T.`